src/pudl/analysis/mcoe.py

```diff
@@ -99,7 +99,7 @@ def heat_rate_by_gen(pudl_out):
                                'plant_id_eia',
                                'unit_id_pudl',
                                'generator_id']].drop_duplicates()
-    gens_simple = pd.merge(gens_simple, bga_gens,
+    bga_gens = pd.merge(gens_simple, bga_gens,
                            on=['report_date', 'plant_id_eia', 'generator_id'],
                            validate='one_to_one')
```

Comment:
Hey @grgmiller thanks for the PR! I agree this does look pretty odd. I can't recall if it was @cmgosnell or I who wrote this part.

I'm not 100% sure, and I'm playing with it a little bit now, but I could *imagine* the purpose of this line being the `validate='one_to_one'` -- which will cause the merge to fail if the merge keys are not unique in both dataframes.  However, in this case we just did `drop_duplicates()` on `bga_gens` so of course they have to be unique there.  So the only thing this really does is make sure that there's only a single `fuel_type_code_pudl` associated with each combination of `report_date`, `plant_id_eia` and `generator_id`.  Right?

We may not need this `gens_simple` thing at all, since the `fuel_type_code_pudl` is merged in from `gens_eia860` a little further down. Testing now to see if these result in equivalent dataframes...

Reply:
Okay, it appears that getting rid of the `gens_simple` entirely and merging into `bga_gens` instead generate the same outputs from this function, so I'm inclined to simply remove the `gens_simple` unless you have some better reocllection of what we were trying to do here @cmgosnell

Reply:
I haven't looked into this deeply, but it looks like `gens_simple` is there only to include the `fuel_type_code_pudl` column, but that column is also being merge in below in the chunk of code stating with `hr_by_gen = pudl.helpers.merge_on_date_year`.